examples/rgw/boto3/service-2.sdk-extras.json

```diff
@@ -408,7 +408,40 @@
         },
         "ReadStats":{"type":"boolean"},
         "ObjectCount":{"type":"integer"},
-        "BytesUsed":{"type":"integer"}
+        "BytesUsed":{"type":"integer"},
+        "CreateBucketConfiguration":{
+            "type":"structure",
+            "members":{
+                "BucketIndex":{
+                    "shape":"BucketIndex",
+                    "documentation":"<p>Configuration to customize the bucket index layout.</p> <note> <p>This element is a Ceph RGW extension.</p> </note>"
+                }
+            }
+        },
+        "BucketIndex":{
+            "type":"structure",
+            "required":[
+                "Type"
+            ],
+            "members":{
+                "Type":{
+                    "shape":"BucketIndexType",
+                    "documentation":"<p>Bucket index type: Normal or Indexless.</p>"
+                },
+                "NumShards":{
+                    "shape":"BucketIndexNumShards",
+                    "documentation":"<p>Number of initial bucket index shards. Only valid for the <code>Normal</code> index type.</p>"
+                }
+            }
+        },
+        "BucketIndexType":{
+            "type":"string",
+            "enum":[
+                "Normal",
```

Comment:
Wondering if this is an opportunity to name hashing bucket index types as something other than "Normal". It's hard to know the future, but it's conceivable that a different index type might become more Normal in the not so distant future.

So I wonder if something along the lines of "Hash.v1" might be more useful, specifying that it's using version 1 of some type of Hashed indexing. That would allow other types of hashing in the future as well.

But once we name this type Normal, I fear we're stuck with referring to that forever as Normal, even if it's ultimately rare.

Reply:
the string "Normal" has been used in https://github.com/ceph/ceph/blob/b40036b/src/rgw/rgw_bucket_layout.cc#L22-L42 since pacific, so would be hard to change now

but when `RGWCreateBucket_ObjStore_S3::get_params()` parses this type, it could map some other name to "Normal". maybe the s3 api could call it "ShardUnordered"?